modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java

```diff
@@ -2649,42 +2690,60 @@ else if (cpDefinitionOptionRel.isRequired()) {
 									"checkbox_multiple|radio|select")) {
 
 									boolean multipleSelect = false;
+									Map<String,String> stringMap = new HashMap<>();
 
 									if (Objects.equals(
 										optionTypeKey, "checkbox_multiple")) {
-										//transform string to map
 
-										//then set to true
 										multipleSelect = true;
+										stringMap = JSONUtil.toStringMap(jsonObject);
 									}
 
 									List<CPOptionValue> cpOptionValues =
 										cpOption.getCPOptionValues();
 
-									// check if we have a valid option value
 									for (CPOptionValue cpOptionValue :
 										cpOptionValues) {
 
 										if (multipleSelect) {
-											//check against map
+											if (stringMap.containsValue(
+												cpOptionValue.getKey())) {
 
-											//remove item from map
+												stringMap.remove(
+													"value", cpOptionValue.getKey());
+											}
 
-											//if map is empty set contains requiredOption to true
+											if (!stringMap.containsKey("value")) {
+												containsRequiredOption = true;
+												break;
+											}
 										}
 										else {
 											if (Objects.equals(
 												cpOptionValue.getKey(),
-												value)) {
+												value.get(0))) {
 
 												containsRequiredOption = true;
 												break;
 											}
 										}
 									}
+								} else if (
+									Objects.equals(
+										optionTypeKey, "checkbox")) {
+
+									String valueString = value.getString(0);
+
+									if (Objects.equals(
+										valueString, cpOption.getKey()) ||
+										Objects.equals(valueString, "[]")) {
+
+										containsRequiredOption = true;
+									}
 								}
 								else {
 									//can't validate other form options easily
```

Comment:
Other form options are user-input (e.g. dates, text, etc) so I don't believe there's a way to properly validate these